### DISCUSSION TITLE
Does it support grpc?

### DISCUSSION THREAD

**Discussion by jet10000** [litestar-org/litestar#600] [Ideas]
Are there plans to support grpc?

**Comment by Goldziher** [comment]
Hi. What do you have in mind? gRPC is generated code so I wonder how that could look?

**Comment by jet10000** [comment]
Now starlite supports http and websocket protocols, is it possible to add support for protobuf and grpc?

https://github.com/grpc/grpc

https://github.com/vmagamedov/grpclib

https://github.com/python-trio/purerpc

https://github.com/gluk-w/django-grpc

**Comment by Goldziher** [comment]
starlite is based on the ASGI specification, you can find it here: https://asgi.readthedocs.io/en/latest/

We can create an adapter for GRPC perhaps, I need to understand though how this could look like and what is the use case for this - also if there are any interested users aside from your good self.

**Reply by jet10000** [reply]
That's great

**Comment by jet10000** [comment]
web app

https://connect.build/docs/web/getting-started/

https://buf.build/blog/connect-web-protobuf-grpc-in-the-browser

flutter app

https://grpc.io/docs/languages/dart/quickstart/

machine learning

https://towardsdatascience.com/serving-deep-learning-model-in-production-using-fast-and-efficient-grpc-6dfe94bf9234

proxy grpc web to grpc

https://github.com/mholt/caddy-grpc-web

**Comment by jet10000** [comment]
https://github.com/sinisaos/grpc-python-crud

**Reply by Goldziher** [reply]
I don't see the point. Why use a rest framework alongside protobuf?

**Comment by Tobi-De** [comment]
@jet10000 Grpc is its own framework and they have support for python, see [here](https://grpc.io/docs/languages/python/quickstart/).
You can already build a grpc server with python.
As far as I know most of the time grpc is used for backend to backend communication.
Starlite is also another framework, meant to build json based api's, in my opinion it doesn't really make sense to add "support" for grpc in starlite, I'm not even sure how and what it would look like.

Can you describe your use case a bit more? What are you trying to do with grcp in a starlite application or vice versa?

**Reply by winstxnhdw** [reply]
If you've used services like Qdrant and Triton, you'll see that they both support gRPC and REST endpoints. Currently, they achieve this by maintaining both schemas. This is a huge maintenance burden. Fortunately, there's a lot of overlap in the schemas and my guess is that OP wants Litestar to perform gRPC codegen so that he doesn't have to maintain both schemas and can just maintain a single universal schema.

**Reply by winstxnhdw** [reply]
We could probably use [gnostic-grpc](https://github.com/google/gnostic-grpc) to do this. Would Litestar accept a contribution for this? @provinzkraut

**Reply by JacobCoffee** [reply]
This may be best as a [plugin](https://docs.litestar.dev/latest/usage/databases/sqlalchemy/plugins/index.html#) which is what we are moving a lot of core functionality into–out from the stdlib–in v3.

**Reply by winstxnhdw** [reply]
Sure, but what I’d really like to know is that given the case where the plugin is unable to convert a schema to an equivalent Protobuf and we’re forced to only provide limited gRPC support, is this something the Litestar plugin ecosystem would accept?
